pages/collection/[id]/play/index.js

```diff
@@ -0,0 +1,265 @@
+  }
+  function handleConfirmStop() {
+    router.push(`/collection/${id}`);
+  }
+
+  // Nav Functions
+
+  function handleNext() {
+    if (currentPage < totalPages - 1) {
+      setShowAnswer(false);
+      // navigate to new URL while keeping current page structure
+      router.push({
+        pathname: router.pathname, // ensure to stay on same page
+        query: { id, card: currentPage + 1 }, // go to next card
+      });
+    } else {
+      setShowFinalMessage(true);
+    }
+  }
+
+  // Retry Function
+  function handleRetry() {
+    setShowAnswer(false);
+    setShowFinalMessage(false);
+    setIsCorrect(0);
+    setTimeElapsed(0); // Reset the timer
+    router.push(`/collection/${id}/play?card=0`);
+  }
+
+  //Correct
+  function handleCorrect() {
+    setIsCorrect(isCorrect + 1);
+    handleNext();
+  }
+
+  return (
+    <StyledMain color={backgroundColor}>
+      <StyledHeader>
+        <StyledHeadline>{currentCollection.title}</StyledHeadline>
+        {showStopConfirm ? (
+          <StyledButton onClick={handleToggle} stop={stop}>
+            ⏹ stop
+          </StyledButton>
+        ) : (
+          <>
+            <StyledButtonContainer>
+              <StyledButton onClick={handleConfirmStop}>confirm</StyledButton>
+              <StyledButton onClick={handleToggle}>cancel</StyledButton>
+            </StyledButtonContainer>
+          </>
+        )}
+      </StyledHeader>
+
+      {showFinalMessage ? (
+        <StyledMessageContainer>
+          <h2>Well done!</h2>
+          <p>
+            {isCorrect} / {totalPages} answered correctly
+          </p>
+          <p>Time spent: {formatTime(timeElapsed)}</p>
+          <StyledButtonContainer>
+            <StyledButton onClick={handleRetry}>retry</StyledButton>
+            <StyledButton onClick={handleConfirmStop}>quit</StyledButton>
+          </StyledButtonContainer>
+        </StyledMessageContainer>
+      ) : (
+        <>
+          {isCounting ? (
+            <StyledCountDown>{countDown}</StyledCountDown> // Show countdown before quiz starts
+          ) : (
+            <>
+              <StyledCardContainer>
+                <p>Time: {formatTime(timeElapsed)}</p>
+                {filteredFlashCards.length > 0 ? (
+                  <>
+                    <PlayModeCard
+                      card={filteredFlashCards[currentPage]}
+                      showAnswer={showAnswer}
+                      setShowAnswer={setShowAnswer}
+                    />
+                  </>
+                ) : (
+                  <p>No cards found.</p>
+                )}
+              </StyledCardContainer>
+
+              <StyledFooter>
+                <button onClick={handleNext}>❌</button>
+                <p aria-label="page-counter">
+                  {currentPage + 1} / {totalPages}
+                </p>
+
+                <button onClick={handleCorrect}>✅</button>
+              </StyledFooter>
+            </>
+          )}
+        </>
```

Comment:
the three game states finalMessage, countdown and play mode are displayed by using nested ternary operators. Instead please use a simpler structure: 
```jsx
const [gameState, setGameState] = useState("countdown") // countdown, end or playing

...

{gameState === 'countdown' && <> html for displaying countdown </>}
{gameState === 'playing' && <> html for displaying end message </>}
{gameState === 'end' && <> html for displaying final message </>}


```